Makefile

```diff
@@ -117,6 +117,13 @@ generate-crds: $(KUSTOMIZE) manifests kustomize ## generate final crds with kust
 	mkdir -p generated-crds
 	$(KUSTOMIZE) build config/crd -o generated-crds # If -o points to a folder, kustomize saves them as several files instead of 1
 
+.PHONY: tag
+tag:
+	@git tag "${TAG}" || (echo "Tag ${TAG} already exists. If you want to retag, delete it manually and re-run this command" && exit 1)
+	@git-chglog --output CHANGELOG.md
+	@git commit -m 'Update CHANGELOG.md' -- CHANGELOG.md
+	@git tag -f "${TAG}"
```

Comment:
While generating chglog + tagging is nice, this means that the tag will be created for a code that is still to be reviewed and merged.

It's still also dependent on how people push, if only the code, or they are also pushing the tag.

Also, the PR of updating the changelog will not be included in the generated changelog. Which means that on next generation, it will pop-up (as it does now).

If we are happy with tag creation in unmerged code, we should change the github actions release workflow so it only runs when there's a tag, and that tag is merged in main, to not release unreviewed/unmerged code by mistake.

Reply:
I temporarily removed the `make tag` target. We can add it back once we sort out the issue with the changelog generator